src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -239,7 +251,7 @@ func (r *ClusterReconciler) createExternalNodesList(
 		return []string{}, fmt.Errorf("failed to retrieve node port service %s: %w", nodePortName, err)
 	}
 
-	if len(nodePortSvc.Spec.Ports) != 1 || nodePortSvc.Spec.Ports[0].NodePort == 0 {
+	if len(nodePortSvc.Spec.Ports) != 2 || nodePortSvc.Spec.Ports[0].NodePort == 0 || nodePortSvc.Spec.Ports[1].NodePort == 0 {
```

Comment:
this is not ideal, we do this per port-name but we always check both ports here, also we pull the nodes for each port name but the ports actually run in the same process so always share the port. We should refactor this to reflect that

Reply:
I took care of this https://github.com/vectorizedio/redpanda/pull/1021/commits/7f8c320d66faddfd601cbc0edc77769acae34917